app/models/health_care_application.rb

```diff
@@ -13,7 +13,9 @@ class HealthCareApplication < ApplicationRecord
   validates(:state, presence: true, inclusion: %w[success error failed pending])
   validates(:form_submission_id_string, :timestamp, presence: true, if: :success?)
 
-  after_save :send_failure_mail, if: proc { |hca| hca.state_changed? && hca.failed? && hca.parsed_form&.dig('email') }
+  after_save(:send_failure_mail, if: proc do |hca|
```

Comment:
Can we extract this into a separate PR? Looks like it's not Rails 5.1 specific

Reply:
`saved_change_to_attribute` does not work in rails 5.0